File: Competitive Programming 3/Section 1.4/Game (Card)/11678 - Cards' Exchange/Python/main.py
def main():
    while True:
        a,b=map(int,input().split())
        if a==0 and b == 0:
            break
        pointer,amountA,amountB=0,0,0
        nums=input().split()
        aArr=[int(tk) for i,tk in enumerate(nums) if tk!=nums[i-1]]
        last=0
        bArr=map(int,input().split())
        for x in bArr:
            if x != last:
                while pointer<len(aArr) and aArr[pointer] < x:
                    pointer+=1
                    amountA+=1
                if pointer == len(aArr) or aArr[pointer] !=x:
                    amountB+=1
                if pointer < len(aArr) and aArr[pointer] == x:
                    pointer+=1
                last = x
        while pointer < len(aArr):
            pointer+=1
            amountA+=1
        print(min(amountA,amountB))
main()

File: Competitive Programming 3/Section 1.4/Game (Card)/11678 - Cards' Exchange/Java/Main.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    static Scanner s = new Scanner(System.in);
    static ArrayList<Integer> aArr;
    public static void main(String[]args){
        while(true){
            aArr = new ArrayList<Integer>();
            int a=s.nextInt(),b=s.nextInt(),pointer = 0,amountB=0,amountA=0,last=0;
            if(a==0&&b==0)
                break;
            for(int i = 0 ; i < a ; i ++){
                int x = s.nextInt();
                if(last!=x){
                    aArr.add(x);
                    last = x;
                }
            }
            last=0;
            for(int i = 0 ; i < b ; i ++){
                int x = s.nextInt();
                if(x!=last){
                    for(;pointer<aArr.size()&&aArr.get(pointer)<x;pointer++){
                        amountA++;
                    }
                    if(pointer==aArr.size()||aArr.get(pointer)!=x){
                        amountB++;
                    }
                    if(pointer<aArr.size()&&aArr.get(pointer)==x)
                        pointer++;
                    last=x;
                }
            }
            for(;pointer<aArr.size();pointer++){
                    amountA++;
            }
            System.out.printf("%d\n",Math.min(amountA,amountB));
        }
    }
}